tests/utils/http_server.py

```diff
@@ -52,6 +52,18 @@ def chunked_drip(handler):
     handler.wfile.write('0\r\n\r\n'.encode('utf-8'))
 
 
+@TestHandler.handler('GET', '/html')
+def basic_html_page(handler):
+    content = b"\x00<some></some>"
+
+    handler.send_response(200)
+    handler.send_header('Content-Type', 'text/html; charset=utf-8')
+    handler.send_header('Content-Length', len(content))
+    handler.end_headers()
+
+    handler.wfile.write(content)
```

Comment:
You could just use the `httpbin` fixture (`httpbin + '/html'`).

```
$ http pie.dev/html
HTTP/1.1 200 OK
CF-Cache-Status: DYNAMIC
CF-RAY: 6be0bf830dce2780-PRG
Connection: keep-alive
Content-Encoding: gzip
Content-Type: text/html; charset=utf-8
```

Reply:
I wanted to ensure the mime type gets matched by converters but unfortunately I was not able to modify httpbin's html endpoint's data (converts only get activated when the data includes `\x00`).

Reply:
I think you could simply check if there’s some color in there:

```python
from .utils import http, COLOR


def test_html(httpbin):
    r = http('--body', httpbin + '/html')
    assert COLOR in r
````

Reply:
There’s usually a way to test pretty anything just against `httpbin` :) None of those mocking tests have been added by me.

Here, you might actually also need to provide a custom `Environment` to enable colorized output (`stdout_isatty=True`)